From 27cf367ecf405760fd71795e9b5f07a077ce1b9e Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Thu, 10 Dec 2015 08:11:43 -0800
Subject: [PATCH 1/2] ld -r doesn't need plugin for slim lto object

Plugin isn't required on slim lto object for relocatable link.

	PR ld/19317
	* symtab.cc (Symbol_table::add_from_relobj): Don't complain
	plugin needed to handle slim lto object for relocatable link.
---
 gold/symtab.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gold/symtab.cc b/gold/symtab.cc
index 22a44e0..f43fb4f 100644
--- a/gold/symtab.cc
+++ b/gold/symtab.cc
@@ -1171,7 +1171,8 @@ Symbol_table::add_from_relobj(
 
       const char* name = sym_names + st_name;
 
-      if (strcmp (name, "__gnu_lto_slim") == 0)
+      if (!parameters->options().relocatable()
+	  && strcmp (name, "__gnu_lto_slim") == 0)
         gold_info(_("%s: plugin needed to handle lto object"),
 		  relobj->name().c_str());
 
-- 
2.5.0

